### ISSUE TITLE
Provide data attribution capabilities.

### ISSUE THREAD

**Issue by pelson** [SciTools/cartopy#32]
Many of the datasets used by cartopy require attribution. Provide an interface which makes it easy to attribute a dataset which is automatically displayed at the bottom of a map.

**Comment by QuLogic** [comment]
> standard matplotlib interface

I guess [this API](http://matplotlib.org/users/annotations_guide.html#placing-artist-at-the-anchored-location-of-the-axes) would work here?

**Comment by pelson** [comment]
> I guess this API would work here?

Yep. There is a complexity in bubbling up the attributions from the source datasets e.g.:

```
ax.add_wms(USGS_wms)
```

Should automatically add the annotation to the axes (it should be possible to remove as a user though). What makes things a little harder is that the attribution changes as you look at different areas/zoom levels even within the same data source - for that reason, I'd be surprised if there wasn't attribution information available from interfaces such as WMS etc.

Complexity aside - there is scope to keep the problem simple and do something as simple as checking whether an object being added to a GeoAxes has a ``attribution`` attribute, and automatically create the attribution box for you.
